File: practice_problems/practice_problems_web/src/api/question.ts
import axios from 'axios';
// 假设你有一个通用的类型定义文件，如果没有，可以删掉这行导入，直接用 any 或在下面定义
import type { ApiResponse } from '../types'; 

const API_URL = 'http://localhost:8080/api/v1/questions';

// --- 类型定义 (建议放在 src/types/index.ts 中，这里为了方便先写在这里) ---

// 创建题目的参数结构 (对应后端 CreateQuestionRequest)
export interface CreateQuestionParams {
  knowledgePointId: number;
  questionText: string;
  option1: string;
  option2: string;
  option3: string;
  option4: string;
  correctAnswer: number; // 1, 2, 3, 4
  explanation: string;
  note?: string;
}

// 题目列表项结构 (对应后端 Question)
export interface QuestionItem {
  id: number;
  knowledgePointId: number;
  questionText: string;
  option1: string;
  option2: string;
  option3: string;
  option4: string;
  correctAnswer: number;
  explanation: string;
  note: string;
  createTime: string;
}

// --- 接口函数 ---

/**
 * 获取某知识点下的题目列表
 * GET /api/v1/questions?point_id=1
 */
export const getQuestions = (pointId: number) => {
    return axios.get<ApiResponse<QuestionItem[]>>(`${API_URL}?point_id=${pointId}`);
}

/**
 * 创建新题目
 * POST /api/v1/questions
 */
export const createQuestion = (data: CreateQuestionParams) => {
    return axios.post<ApiResponse<{id: number}>>(API_URL, data);
}

/**
 * 更新题目 (预留)
 * PUT /api/v1/questions/:id
 */
export const updateQuestion = (id: number, data: Partial<CreateQuestionParams>) => {
    return axios.put<ApiResponse<null>>(`${API_URL}/${id}`, data);
}

/**
 * 删除题目
 * DELETE /api/v1/questions/:id
 */
export const deleteQuestion = (id: number) => {
    return axios.delete<ApiResponse<null>>(`${API_URL}/${id}`);
}
export const getQuestionsByCategory = (categoryId: number) => {
  // 假设后端有这个接口，URL 类似 /api/v1/questions?category_id=xxx
  // 如果后端没有，你可能需要前端循环调用 getQuestions(pointId) 来拼接，但建议后端加接口
  return axios.get<ApiResponse<QuestionItem[]>>(`${API_URL}?category_id=${categoryId}`);
}

File: model/question.go
package model

// Question 对应新的 questions 表
type Question struct {
	ID               int    `json:"id"`
	KnowledgePointID int    `json:"knowledgePointId"` // 对应 knowledge_point_id
	QuestionText     string `json:"questionText"`     // 对应 question_text

	Option1    string `json:"option1"`
	Option1Img string `json:"option1Img"`
	Option2    string `json:"option2"`
	Option2Img string `json:"option2Img"`
	Option3    string `json:"option3"`
	Option3Img string `json:"option3Img"`
	Option4    string `json:"option4"`
	Option4Img string `json:"option4Img"`

	CorrectAnswer int    `json:"correctAnswer"` // 1, 2, 3, 4
	Explanation   string `json:"explanation"`
	Note          string `json:"note"`

	CreateTime string `json:"createTime"`
	UpdateTime string `json:"updateTime"`
}

// CreateQuestionRequest 创建请求
type CreateQuestionRequest struct {
	KnowledgePointID int    `json:"knowledgePointId" binding:"required"`
	QuestionText     string `json:"questionText" binding:"required"`

	Option1    string `json:"option1"`
	Option1Img string `json:"option1Img"`
	Option2    string `json:"option2"`
	Option2Img string `json:"option2Img"`
	Option3    string `json:"option3"`
	Option3Img string `json:"option3Img"`
	Option4    string `json:"option4"`
	Option4Img string `json:"option4Img"`

	CorrectAnswer int    `json:"correctAnswer" binding:"required"` // 必填
	Explanation   string `json:"explanation"`
}

// UpdateQuestionRequest 更新请求
type UpdateQuestionRequest struct {
	QuestionText string `json:"questionText"`

	Option1    string `json:"option1"`
	Option1Img string `json:"option1Img"`
	Option2    string `json:"option2"`
	Option2Img string `json:"option2Img"`
	Option3    string `json:"option3"`
	Option3Img string `json:"option3Img"`
	Option4    string `json:"option4"`
	Option4Img string `json:"option4Img"`

	CorrectAnswer int    `json:"correctAnswer"`
	Explanation   string `json:"explanation"`
}

type UpdateNoteRequest struct {
	QuestionID int    `json:"question_id" binding:"required"`
	Note       string `json:"note"` // 允许为空，为空可能意味着清空备注
}
